modules/internal/ChapelIO.chpl

```diff
@@ -524,73 +524,35 @@ module ChapelIO {
   }
   
   class StringWriter: Writer {
-    var s: c_string_copy; // Should be initialized to NULL.
-    proc StringWriter(x:c_string) {
-      this.s = __primitive("string_copy", x);
+    var s: string; // Should be initialized to NULL.
+    proc StringWriter(x:string) {
+      this.s = x;
     }
     proc writePrimitive(x) {
-      // TODO: Implement += so it consumes a c_string_copy LHS.
-      var aug = x:c_string_copy;
-      this.s += aug;      // The update frees this.s before overwriting it.
-      chpl_free_c_string_copy(aug);
-    }
-    proc ~StringWriter() {
-      chpl_free_c_string_copy(this.s);
-      __primitive("=", this.s, _nullString);
+      this.s += x:string;
     }
   }
   
-  // Convert 'x' to a string just the way it would be written out.
-  // Includes Writer.write, with modifications (for simplicity; to avoid 'on').
-  proc _cast(type t, x) where t == c_string_copy {
-    //proc isNilObject(o: object) return o == nil;
-    //proc isNilObject(o) param return false;
-    const w = new StringWriter();
-    //if isNilObject(x) then "nil".writeThis(w);
-    //else                   x.writeThis(w);
-    w.write(x);
-    const result = w.s;
-    __primitive("=", w.s, _nullString);
-    delete w;
-    return result;
-  }
-  
   pragma "dont disable remote value forwarding"
   proc ref c_string.write(args ...?n) {
-    var sc = new StringWriter(this);
+    compilerError("c_string.write()");
+    //TODO strings: something...
+    /*
```

Comment:
I don't think we should support c_string.write. So I think you should remove the body and just
leave this fn as compilerError.